test/openlr.cc

```diff
@@ -161,6 +168,46 @@ TEST(OpenLR, TooSmallReference) {
   EXPECT_THROW(auto locRef = LineLocation(decode64(location)), std::invalid_argument);
 }
 
+TEST(OpenLR, CreateLinearReference) {
+  // make a reference directly using lrps
+  std::vector<PointLL> points{{-76.55015772558436, 40.482238379871575},
+                              {-76.55060287976856, 40.48275880818994},
+                              {-76.55108895828207, 40.489983794570605},
+                              {-76.54576249703334, 40.49195968608316}};
+  std::vector<LocationReferencePoint> lrps;
+  unsigned char frc = 0;
+  auto fow = LocationReferencePoint::FormOfWay::MOTORWAY;
+  unsigned char lowest_frc_next_point = 7;
+  uint16_t bearing;
+  for (const auto& p : points) {
+    // frc and fow are 3 bits wide, we dont use the last 2 bits for anything
+    unsigned char attribute1 = (frc << 3) | fow;
+    // first or intermediate point
+    if (&p != &points.back()) {
+      bearing = static_cast<uint16_t>(p.Heading(*std::next(&p)));
+      auto distance = p.Distance(*std::next(&p));
+      lrps.emplace_back(p.lng(), p.lat(), bearing, frc, fow, lrps.empty() ? nullptr : &lrps.back(),
+                        distance, lowest_frc_next_point);
+    } // last point
+    else {
+      bearing += 180;
+      bearing %= 360;
+      lrps.emplace_back(p.lng(), p.lat(), bearing, frc, fow, lrps.empty() ? nullptr : &lrps.back());
+    }
+    // try different frcs and fows for kicks
+    ++frc;
+    fow = static_cast<LocationReferencePoint::FormOfWay>(static_cast<uint8_t>(fow) + 1);
+    --lowest_frc_next_point;
+  }
+  LineLocation line_location(lrps, 0, 0);
```

Comment:
this is an example of how you can create an openlr record directly. we already had the method to serialize the record to string but we were lacking the ability to create one from raw data.